### DISCUSSION TITLE
NonInteractive mode when using System.Management.Automation

### DISCUSSION THREAD

**Discussion by goffries** [PowerShell/PowerShell#16814] [Q&A]
Hi,

I was just wondering if there is a way to ensure that prompts are suppressed when running a script using System.Management.Automation inside a C# application? Similar to the -NonInteractive  flag you can pass to powershell.exe.

Any help would be appreciated.

**Comment by jborean93** [comment]
Any interaction will not be allowed like in `-NonInteractive` if you do not pass in a host when creating a Runspace or Pipeline. You could even just pass in an implemented [PSHost](https://docs.microsoft.com/en-us/dotnet/api/system.management.automation.host.pshost?view=powershellsdk-7.0.0) value but will need to make sure the [UI](https://docs.microsoft.com/en-us/dotnet/api/system.management.automation.host.pshost.ui?view=powershellsdk-7.0.0) property does not contain a value as this is the layer that handles the user interaction.

For example

```powershell
$ps = [PowerShell]::Create()
$ps.AddScript('Read-Host -Prompt test').Invoke()
[string]$ps.Streams.Error[0]

> A command that prompts the user failed because the host program or the command type does not support user interaction. The host was attempting to request confirmation with the following message:
```

This fails as no host was supplied to handle the user interaction.

```powershell
$settings = [System.Management.Automation.PSInvocationSettings]@{
    Host = $host
}
$ps = [PowerShell]::Create()
$ps.AddScript('Read-Host -Prompt test').Invoke(@(), $settings)

> test: ....
```

This one will actually prompt the user because a PSHost with a UI was supplied.
